### DISCUSSION TITLE
Production well in geothermal doublet

### DISCUSSION THREAD

**Discussion by JustinPogacnik** [idaholab/moose#26802] [Q&A Modules: Porous Flow]
Hi Team,

I'm trying to create a geothermal doublet simulation. I'd like to control both the injection and production well by mass flow rate. The injection well seems to be going fine, but I'm having problems with the production well temperature increasing over time. I've seen some similar discussion and the solution was to use a Peaceman borehole description. I'm not sure how to do that and control the mass flow rate of the production well. It's quite easy to control that flowrate with a pointsource/sink term. But the problem is the temperature at the outflow node. I've tried a few different approaches using either polylinesink or pointsource in both falcon PTGeothermal and PorousFlow physics and seeing the same issue. So I'm pretty well convinced that I'm the problem. Can someone help me out here?

[PRODUCTION_VPO.txt](https://github.com/idaholab/moose/files/14268869/PRODUCTION_VPO.txt)
![Capture](https://github.com/idaholab/moose/assets/10068141/725f401c-f625-4794-8ecc-de2d96b299fb)

**Comment by GiudGiud** [comment]
@cpgr

**Reply by Yulongggggg** [reply]
Hi i have the same issue, have you fixed it?

**Reply by JustinPogacnik** [reply]
No, no one ever responded. I was told by development team to put questions here instead of emailing directly.

**Comment by Yulongggggg** [comment]
Hi i have the same issue, have you fixed it?

**Comment by josebastiase** [comment]
Hi,

I haven't run your code, but a few questions:

1. What happens when you increase the permeability? Still happens?
2. Have you try just to simulate only flux? Does the mass balance match expectations?
 
I think your system is loosing saturation near the well so you may be getting negative pressures, which makes temperature go crazy.

**Reply by JustinPogacnik** [reply]
I've removed everything from my system and moved on with a different code. I only had 20 days last year to investigate this. Got stuck there. If I recall, the pressure looked ok. If you look in the file, I think it's constant water properties. So, no issue with the equations of state creating the issue. I'd be curious if find the solution to this issue.

**Reply by josebastiase** [reply]
If you have the mesh files I can take a look. Thanks

**Reply by lynnmunday** [reply]
You need dirac kernels to remove enthalpy to.  These are dirac kernels I use.

```
# these inject
[DiracKernels]
  [source_1]
    type = PorousFlowPointSourceFromPostprocessor
    variable = porepressure
    mass_flux = mass_flux_src_stage_1
    point = '4.1345165890 220.1299407155 396.4301458718'
    point_not_found_behavior = WARNING
    # block = 100
  []
  [source_1_h]
    type = PorousFlowPointEnthalpySourceFromPostprocessor
    variable = temperature
    mass_flux = mass_flux_src_stage_1
    point = '4.1345165890 220.1299407155 396.4301458718'
    T_in = inject_T
    pressure = porepressure
    fp = tabulated_water
    point_not_found_behavior = WARNING
    # block = 100
  []
[]

# these produce
# PEACEMANS
[Functions]
  [insitu_pp_borehole]
    type = ParsedFunction
    symbol_names = 'back_pressure_Pa'
    symbol_values = '3.44738e6' #500psi back-pressure
    expression = '1.6025e7-${water_weight}*(z-1150)+back_pressure_Pa'
  []
  #delaying production for initial heaviside injection
  [character_function]
    type = PiecewiseLinear
    xy_data = "0 0
    ${fparse production_delay} 0
    ${fparse production_delay + 14500} 1"
  []
[]

[UserObjects]
  [borehole_fluid_outflow_mass]
    type = PorousFlowSumQuantity
  []
  [borehole_prod_temperature]
    type = PorousFlowSumQuantity
  []
[]

[DiracKernels]
  [withdraw_fluid]
    type = PorousFlowPeacemanBorehole
    variable = porepressure
    bottom_p_or_t = insitu_pp_borehole
    SumQuantityUO = borehole_fluid_outflow_mass
    point_file = peaceman_production_points.txt
    function_of = pressure
    fluid_phase = 0
    unit_weight = '0 0 -${water_weight}'
    use_mobility = true
    character = character_function
    point_not_found_behavior = WARNING
  []
  [bh_energy_flow]
    type = PorousFlowPeacemanBorehole
    variable = temperature
    bottom_p_or_t = insitu_pp_borehole
    SumQuantityUO = borehole_prod_temperature
    point_file = peaceman_production_points.txt
    function_of = pressure
    fluid_phase = 0
    unit_weight = '0 0 -${water_weight}'
    use_mobility = true
    use_enthalpy = true
    character = character_function
    point_not_found_behavior = WARNING
  []
[]

[Postprocessors]
  [fluid_report]
    type = PorousFlowPlotQuantity
    uo = borehole_fluid_outflow_mass
  []
  [energy_prod]
    type = PorousFlowPlotQuantity
    uo = borehole_prod_temperature
  []
[]

**Reply by Yulongggggg** [reply]
Hi, since i am using vertical borehole for geothermal, i see the peacemen borehole is for vertical boreholes. Can i use the code below?:

[Postprocessors]
  [A_inj]
    type = AreaPostprocessor
    boundary = injection
    execute_on = initial
  []
  [A_prod]
    type = AreaPostprocessor
    boundary = production
    execute_on = initial
  []
  [inj_flux]
    type = FunctionValuePostprocessor
    function = inj_flux
    execute_on = 'initial timestep_begin'
  []
  [power_prod]
    type      = SideIntegralVariablePostprocessor
    boundary  = production
    variable  = heat_outflow          # W/m²
    execute_on = 'TIMESTEP_END'
  []
[]

[BCs]
  [m_in]
    type = PorousFlowSink
    boundary = injection
    variable = porepressure
    flux_function = inj_flux
    fluid_phase = 0
  []
  [in_T]
    type = PorousFlowEnthalpySink
    variable = temperature
    boundary = injection
    T_in = 300
    fp = water
    flux_function = inj_flux
    porepressure_var = porepressure
  []
  [m_out]
    type = PorousFlowSink
    boundary = production
    variable = porepressure
    flux_function = prod_flux
    fluid_phase = 0
    use_mobility = False
    use_relperm = False
  []
  [m_out_heat]
    type = PorousFlowSink
    boundary = production
    variable = temperature
    flux_function = prod_flux
    fluid_phase = 0
    use_enthalpy = true
    save_in = heat_outflow
  []
[]

**Reply by lynnmunday** [reply]
I don't know.  You will have to read through the documentation and all of the examples that are given in the porous flow documentation.
